timesketch/api/v1/resources/archive.py

```diff
@@ -425,53 +444,181 @@ def _export_view(self, view, sketch, zip_file):
+                sketch.id,
+                "; ".join(error_details),
+            )
+            abort(
+                HTTP_STATUS_CODE_INTERNAL_SERVER_ERROR,
+                "Failed to unarchive sketch. One or more OpenSearch indices could not "
+                f"be opened. Details: {'; '.join(error_details)}",
             )
 
+        # If all indices are open, update the database statuses.
         sketch.set_status(status="ready")
+        logger.info("Sketch %s status set to 'ready'.", sketch.id)
 
-        indexes_to_open = []
+        # Set all timelines in the sketch to ready.
         for timeline in sketch.timelines:
             if timeline.get_status.status != "archived":
                 continue
             timeline.set_status(status="ready")
-            search_index = timeline.searchindex
+            logger.info(
+                "Timeline %d ('%s') status set to 'ready'.",
+                timeline.id,
+                timeline.name,
+            )
+
+        for search_index in successfully_opened_indexes:
             search_index.set_status(status="ready")
-            indexes_to_open.append(search_index.index_name)
+            logger.info(
+                "SearchIndex DB object %s (ID: %s) status updated to 'ready'.",
+                search_index.index_name,
+                search_index.id,
+            )
 
-        # TODO (kiddi): Move this to lib/datastores/opensearch.py.
-        if indexes_to_open:
-            try:
-                self.datastore.client.indices.open(",".join(indexes_to_open))
-            except opensearchpy.NotFoundError:
-                logger.error(
-                    "Unable to open index, not found: {:s}".format(
-                        ",".join(indexes_to_open)
-                    )
-                )
+        # Also set any non-archived search indexes to ready, just in case.
+        all_sketch_search_indexes = {
+            tl.searchindex for tl in sketch.timelines if tl.searchindex
+        }
+        for search_index in all_sketch_search_indexes:
+            if search_index.get_status.status != "ready":
+                search_index.set_status(status="ready")
+
+        db_session.commit()
+        logger.info("Unarchiving of sketch %s complete.", sketch.id)
 
         return HTTP_STATUS_CODE_OK
 
     def _archive_sketch(self, sketch: Sketch):
-        """Unarchives a sketch by opening up all indices and removing labels.
+        """Archives a sketch. This involves:
+
+        1. Setting the sketch status to 'archived'.
+        2. For each SearchIndex associated with the sketch:
+          the system attempts to close the corresponding OpenSearch index.
+           - If the OpenSearch index is successfully closed, the SearchIndex's
+             database status is set to 'archived'.
+           - If the OpenSearch index is not found, it's considered effectively
+             closed, and the SearchIndex's database status is set to 'fail'.
```

Comment:
In context of your action in line 584 above, wouldn't this lead to those searchindex being set as ready in the db again when a sketch gets unarchived?